feat: add permutator that can permutate over the number type

janechu opened the pull request:
<!--- Provide a summary of your changes in the title field above. For guidance on formatting, see the comment at the bottom of this template. -->

# Description

<!--- Describe your changes. -->
This kicks off the work to add the permutator, a utility written to provide permutations over JSON schema. This is the initial setup and iteration over the number type. Future additions include the types `string`, `null`, `boolean`, `array`, `object` and `integer`.   

## Issue type checklist

<!--- What type of change are you submitting? Put an x in the box that applies: -->

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

<!--- If yes, describe the impact. -->

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

<!-- Do your changes add or modify components in the @microsoft/fast-components package? Put an x in the box that applies: -->

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

<!--- Review the list and check the boxes that apply. -->

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

<!---
Formatting guidelines:

Accepted peer review title format:
<type>: <description>

Example titles:
    chore: add unit tests for all components
    feat: add a border radius to button
    fix: update design system to use 3px border radius

    <type> is required to be one of the following:

        - chore: A change that does not impact distributed packages.
        - fix: A change which fixes an issue.
        - feat: A that adds functionality.

    <description> is required for the CHANGELOG and speaks to what the user gets from this PR:

        - Be concise.
        - Use all lowercase characters. 
        - Use imperative, present tense (e.g. `add` not `adds`.)
        - Do not end your description with a period.
        - Avoid redundant words.

For additional information regarding working on FAST, check out our documentation site:
https://www.fast.design/docs/en/contributing/working
-->

nicholasrice commented on packages/tooling/fast-tooling-wasm/permutator/build.js:
```diff
@@ -0,0 +1,55 @@
+    `${changeDir} && emcc ${files} -Os ${settings} -o permutator.js`,
+    (error, stdout, stderr) => {
+        if (error) {
+            console.log(`error: ${error.message}`);
```
Should this throw / exit non-zero since it errored?

scomea commented on packages/tooling/fast-tooling-wasm/app/index.ts:
```diff
@@ -1,20 +1,15 @@
-function getPemutation() {
-    console.log("Expect 10: ", permutate(5));
+function receivePermutation(e) {
+    console.log("permutation recieved", e);
```
nit: "received"

scomea commented on packages/tooling/fast-tooling-wasm/permutator/permutate.c:
```diff
@@ -0,0 +1,51 @@
+                        )
+                    );
+                    break;
+                case String:
```
Do we need to enumerate all the other cases?

janechu replied:
Yes, C will not compile otherwise. Kind of one of the benefits of creating an enum actually.

janechu commented:
@EisenbergEffect I've already added build instructions in this projects CONTRIBUTING.md file, but I've added the other comments and more context to the README.md.
